backend/src/routes/v2/model/file/getDownloadFile.ts

```diff
@@ -25,9 +36,15 @@ interface GetDownloadFileResponse {
 export const getDownloadFile = [
   bodyParser.json(),
   async (req: Request, res: Response<GetDownloadFileResponse>) => {
-    const {
-      params: { fileId },
-    } = parse(req, getDownloadFileSchema)
+    const { params } = parse(req, getDownloadFileSchema)
+
+    let fileId: string
+    if ('semver' in params) {
+      const file = await getFileByReleaseFileName(req.user, params.modelId, params.semver, params.fileName)
+      fileId = String(file._id)
+    } else {
+      fileId = params.fileId
+    }
 
     const file = await getFileById(req.user, fileId)
```

Comment:
If a semver is provided, will this be the second time you retrieve the same file?